org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/batch/ClasspathDirectory.java

```diff
@@ -360,7 +358,7 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
 @Override
 public void reset() {
 	super.reset();
-	this.directoryCache = new Hashtable(11);
+	this.directoryCache.clear();
```

Comment:
Do you know why teh secondaryTypes are not cleared?

Reply:
I also wondered about that. i decided to only make a performance change here keeping logic as is.

Reply:
@mpalat : any clue? `packageSecondaryTypes` field was added in c39467a5717299e0743e249af795ac84f7541935

Reply:
> @mpalat : any clue? `packageSecondaryTypes` field was added in [c39467a](https://github.com/eclipse-jdt/eclipse.jdt.core/commit/c39467a5717299e0743e249af795ac84f7541935)

I believe its a bug - @jukzi  could you please add code to clear this also in reset alongwith this PR.

Reply:
will do